pallets/genesis-history/src/migration.rs

```diff
@@ -0,0 +1,71 @@
+// This file is part of pallet-genesis-history
+
+// Copyright (C) 2020-2023  Intergalactic, Limited (GIB).
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License..
+
+pub use crate::*;
+pub use frame_support::{
+	log,
+	traits::{Get, StorageVersion},
+	weights::Weight,
+};
+
+// This migration fixes the corrupted value in the storage and fixes it
```

Comment:
so the current values are not correct?

Reply:
Yes. We manually set the storage value to something, but the storage has different data type.